gtsfm/utils/geometry_comparisons.py

```diff
@@ -347,6 +347,23 @@ def compute_points_distance_l2(wti1: Optional[Point3], wti2: Optional[Point3]) -
     return np.linalg.norm(wti1 - wti2)
 
 
+def compute_cyclic_rotation_error(i1Ri0: Rot3, i2Ri1: Rot3, i2Ri0: Rot3) -> float:
+    """Computes the cycle error in degrees after composing the three input rotations.
+
+    The cycle error is given by the angle of inv(i2Ri0) * i2Ri1 * i1Ri0.
+
+    Args:
+        i1Ri0: Relative rotation of first edge.
+        i2Ri1: Relative rotation of second edge.
+        i2Ri0: Relative rotation of third edge.
```

Comment:
prefer telling edge from camera i1 to camera i2 and so on as we have not defined edges.